### PR TITLE
[WIP] Static Elementary Types Introduced

### LINKED ISSUE #4502: Introduce static elementary types to use
Code like `IntegerType(160, IntegerType::Modifier::Address);` is very common and should be replaced by something like `ElementaryTypes::Address` (i.e. ElementaryTypes should be a struct that contains several static members initialized to specific instances of the types) at least for the more common elementary types to improve readability.

Note that the Type class inherits from `std::enabled_shared_from_this` and thus is always a `shared_ptr`. The elements of the `ElementaryTypes` struct should be the types themselves, but this has to be checked during implementation what is more common.

The elementary types should be:

`bytes32`, `byte`, `address`, `uint256` and others that are found in the code.

### PR DESCRIPTION
**Pull request by vaibhavchellani**
### Your checklist for this pull request
Fixes #4502 
Please review the [guidelines for contributing](http://solidity.readthedocs.io/en/latest/contributing.html) to this repository.

Please also note that this project is released with a [Contributor Code of Conduct](CONDUCT.md). By participating in this project you agree to abide by its terms.

### Checklist
- [ ] Code compiles correctly
- [ ] All tests passing
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary
- [ ] Used meaningful commit messages

### Description
Please explain the changes you made here.
Added struct called Elementary Types to replace the frequently used `IntegerType(160, IntegerType::Modifier::Address);` and `ArrayType(DataLocation::Memory, true)` etc , they can be directly referenced like `ElementaryTypes::Address` making it more readable .
Thank you for your help!

### CODE REVIEW

File: libsolidity/ast/Types.h [hunk 1]
```diff
@@ -355,6 +355,19 @@ class IntegerType: public Type
...
+{
+	static IntegerType Address;
+	// static const ArrayType BytesMemory;
+	// static const ArrayType StringMemory;
+	// ElementaryTypes()
+	// {
+	// 	Address = IntegerType(160, IntegerType::Modifier::Address);
+	// };
+};
+IntegerType ElementaryTypes::Address = IntegerType(160, IntegerType::Modifier::Address);
```
Comment (axic) [inline libsolidity/ast/Types.h:368]: This actually needs to be used in the code.

Also it shouldn't be instantiated in the header.

File: libsolidity/ast/Types.h [hunk 2]
```diff
@@ -355,6 +355,19 @@ class IntegerType: public Type
...
 
+struct ElementaryTypes
+{
+	static IntegerType Address;
+	// static const ArrayType BytesMemory;
+	// static const ArrayType StringMemory;
+	// ElementaryTypes()
+	// {
+	// 	Address = IntegerType(160, IntegerType::Modifier::Address);
+	// };
```
Comment (axic) [inline libsolidity/ast/Types.h:366]: These seem to be stray comments.

**Comment by axic** [comment]
@vaibhavchellani are you still working on this?

**Comment by vaibhavchellani** [comment]
@axic yes ill push the updates

**Comment by axic** [comment]
@vaibhavchellani closing this now, but please let us know to reopen if you plan to push to it.

**Pull request closed without merging**